master/getting-started/bare-metal/bare-metal.md

```diff
@@ -427,3 +430,284 @@ an ingress rule allowing access *to* port 999 and an egress rule allowing
+  instead via a LoadBalancer IP that is routable from outside the cluster and
+  maps to one of just a few 'ingress' nodes.  (The LoadBalancer IP is a virtual
+  IP that, at any given time, gets routed somehow to one of those 'ingress'
+  nodes.)
+
+- For a few Services, on the other hand, there is no LoadBalancer IP set up, so
+  those Services should be accessible through their NodePorts on any node.
+
+- All other incoming traffic from outside the cluster should be disallowed.
+
+![]({{site.baseurl}}/images/bare-metal-example.png)
+
+For each Service in the first set, we want to allow traffic from outside the
+cluster that is addressed to `<service-load-balancer-ip>:<service-port>`, but
+only when it enters the cluster through one of the 'ingress' nodes.  For each
+Service in the second set, we want to allow traffic from outside the cluster
+that is addressed to `<node-ip>:<service-node-port>`, via any node.
+
+We can do this by applying Calico pre-DNAT policy to the external interfaces of
+each cluster node.  We use pre-DNAT policy, rather than normal host endpoint
+policy, for two reasons:
+
+1. Normal host endpoint policy is not enforced for incoming traffic to a local
+   pod, whereas pre-DNAT policy is enforced for _all_ incoming traffic.  Here
+   we want to police all incoming traffic from outside the cluster, regardless
+   of its destination, so pre-DNAT is the right choice.
+
+2. We want to express our policy in terms of the external port numbers
+   `<service-port>` and `<service-node-port>`.  The kube-proxy on the ingress
+   node will use DNATs to change those port numbers (and IP addresses) to those
+   of one of the pods that backs the relevant Service.  Our policy therefore
+   needs to be enforced _before_ those DNATs, and of course that is exactly
+   what pre-DNAT policy is for.
+
+Let's begin with the policy to disallow incoming traffic by default.  Every
+outward interface of each node, by which traffic from outside could possibly
+enter the cluster, must be defined as a Calico host endpoint; for example, for
+`eth0` on `node1`:
+
+```
+apiVersion: v1
+kind: hostEndpoint
+metadata:
+  name: node1-eth0
+  node: node1
+  labels:
+    host-endpoint: ingress
+spec:
+  interfaceName: eth0
+```
+
+The nodes that are allowed as load balancer ingress nodes should have an
+additional label to indicate that, let's say `load-balancer-ingress: true`.
+
+Then we can deny all incoming traffic through those interfaces, unless it is
+from a source IP that is known to be within the cluster.  (Note: we are
+assuming that the same interfaces can also be used for traffic that is
+forwarded from other nodes or pods in the cluster - as would be the case for
+nodes with only one external interface.)
+
+```
+apiVersion: v1
+kind: policy
+metadata:
+  name: disallow-incoming
+spec:
+  preDNAT: true
+  order: 100
+  ingress:
+    - action: deny
+      source:
+        notNets: [<pod-cidr>, <cluster-internal-node-cidr>, ...]
```

Comment:
This works / is correct, but I'm wondering if a different example using 2 policies to achieve this might be clearer instead:
- First policy is a normal (not preDNAT) policy that you would setup on a node in a k8s cluster if you are using hostEndpoints.  This is the "allow source <pord-cidr>...".  You would do this independent of whether you were trying to lock down nodePorts.
- Second policy is a preDNAT policy that locks down the nodePorts "deny dest <node-port-range>".

WDYT?